### ISSUE TITLE
dec_inp是不是没用到？

### ISSUE THREAD

**Issue by guilunmeibeautiful** [thuml/TimesNet#12]

**Comment by guilunmeibeautiful** [comment]
def forecast(self, x_enc, x_mark_enc, x_dec, x_mark_dec):#batch_x, batch_x_mark, dec_inp, batch_y_mark
        # Normalization from Non-stationary Transformer
        means = x_enc.mean(1, keepdim=True).detach()
        x_enc = x_enc - means
        stdev = torch.sqrt(
            torch.var(x_enc, dim=1, keepdim=True, unbiased=False) + 1e-5)
        x_enc = x_enc/stdev
       
        # embedding
        enc_out = self.enc_embedding(x_enc, x_mark_enc)  # [B,T,C]
        
        enc_out = self.predict_linear(enc_out.permute(0, 2, 1)).permute(
            0, 2, 1)  # align temporal dimension
        
        # TimesNet
        for i in range(self.layer):
            enc_out = self.layer_norm(self.model[i](enc_out))#每个timesblock后都有一个层归一化
        # porject back
        dec_out = self.projection(enc_out)

        # De-Normalization from Non-stationary Transformer
        dec_out = dec_out * \
                  (stdev[:, 0, :].unsqueeze(1).repeat(
                      1, self.pred_len + self.seq_len, 1))
        dec_out = dec_out + \
                  (means[:, 0, :].unsqueeze(1).repeat(
                      1, self.pred_len + self.seq_len, 1))
        return dec_out
没用到为啥还要传过来，

**Comment by guilunmeibeautiful** [comment]
如下是我的命令行参数：
--task_name=long_term_forecast
--is_training=1
--model_id=Time_96Etth1
--model=TimesNet
--data=ETTh1

**Comment by guilunmeibeautiful** [comment]
没有用到use_amp

**Comment by wuhaixu2016** [comment]
您好，这个代码是TimesNet的，但是因为代码库希望可以保持良好的可扩展性，所以输入参数多了dec_inp，在别的模型中会被用到。
